tests/imaging/test_cropset.py

```diff
@@ -0,0 +1,152 @@
+import deepprofiler.imaging.cropset
+import pytest
+import numpy as np
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+import random
+
+@pytest.fixture(scope="function")
+def cropset():
+    test_set_size = 500
+    test_table_size = 500
+    test_crop_shape = [(16,16,3),()]
+    test_target_size = 500
+    return deepprofiler.imaging.cropset.CropSet(test_set_size,
+                                                     test_table_size,
+                                                     test_crop_shape,
+                                                     test_target_size)
+
+def test_init_cropset(cropset):
+    test_set_size = 500
+    test_table_size = 500
+    test_target_size = 500
+    assert cropset.set_size == test_set_size
+    assert cropset.table_size == test_table_size
+    assert cropset.target_size == test_target_size
+    assert np.array_equal(cropset.crops,np.zeros( (test_table_size, 16, 16, 3) ))
+    assert_frame_equal(cropset.labels,pd.DataFrame(data=np.zeros((test_table_size), dtype=np.int32),  columns=["target"]))
+    assert cropset.pointer == 0
+    assert cropset.ready == False
+
+def test_add_crops(cropset):
+    test_table_size = 500
+    test_crops = []
+    test_load = 50
+    np.random.seed(50)
+    for num in range(test_load):
+        test_crops.append(np.random.randint(256, size=(16, 16, 3), dtype=np.uint16))
+    test_crops = np.array(test_crops)
+    test_labels = np.ones( (test_load,2) )
+    test_labels[:,1] = np.array(test_load*[2])
+    cropset.add_crops(test_crops, test_labels)
+    assert cropset.crops.shape == (test_table_size,16,16,3)
+    assert cropset.labels.shape == (test_table_size,1)
+    assert np.array_equal(cropset.crops,np.concatenate((test_crops,np.zeros( (test_table_size-test_load, 16, 16, 3) ))))
+    assert_frame_equal(cropset.labels,pd.DataFrame(data=np.concatenate((np.ones((test_load), dtype=np.int64),np.zeros((test_table_size-test_load), dtype=np.int64))),columns=["target"]))
+    assert cropset.pointer == 50
+    test_load = 500
+    test_crops = []
+    np.random.seed(500)
+    for num in range(test_load):
+        test_crops.append(np.random.randint(256, size=(16, 16, 3), dtype=np.uint16))
+    test_crops = np.array(test_crops)
+    test_labels = np.ones( (test_load,3) )
+    test_labels[:,2] = np.array(test_load*[4])
+    cropset.add_crops(test_crops, test_labels)
+    assert cropset.pointer == 50
+    assert cropset.ready == True
+    assert np.array_equal(cropset.crops,np.concatenate((test_crops[450:500],test_crops[0:450] )))
```

Comment:
Consider setting constants for the numbers you are using in your tests so that values can be changed easily if necessary